handler/src/main/java/io/netty/handler/ssl/SslHandler.java

```diff
@@ -701,10 +703,15 @@ public void read(ChannelHandlerContext ctx) throws Exception {
     @Override
     public void write(final ChannelHandlerContext ctx, Object msg, ChannelPromise promise) throws Exception {
         if (!(msg instanceof ByteBuf)) {
-            promise.setFailure(new UnsupportedMessageTypeException(msg, ByteBuf.class));
-            return;
+            UnsupportedMessageTypeException exception = new UnsupportedMessageTypeException(msg, ByteBuf.class);
+            ReferenceCountUtil.safeRelease(msg);
+            promise.setFailure(exception);
```

Comment:
inline the exception creation here ?

Reply:
I wanted to wait to release until after `msg` was used on the line above. Currently is just uses the `class` but I didn't want to feed it a released object. wdyt?

Reply:
ah right.. good point.